client/web/src/enterprise/user/settings/auth/UserSettingsPermissionsPage.tsx

```diff
@@ -188,11 +189,16 @@ const TableColumns: IColumn<INode>[] = [
         key: 'repository',
         header: 'Repository',
         render: ({ repository }: INode) =>
-            repository && (
+            repository ? (
                 <div key={repository.id} className="py-2">
                     <ExternalRepositoryIcon externalRepo={repository.externalRepository} />
                     <RepoLink repoName={repository.name} to={repository.url} />
                 </div>
+            ) : (
+                <div className="py-2">
+                    <ExternalRepositoryIcon externalRepo={{ serviceType: 'unknown' }} />
+                    You do not have access to this repository
```

Comment:
```suggestion
                    Private repository
```
I think this could easily be confused by a customer as "the user doesn't have access" vs the viewing admin doesn't have access 😬